src/core/AutomatableModel.cpp

```diff
@@ -326,6 +326,11 @@ void AutomatableModel::setValue( const float value )
 }
 
 
+float AutomatableModel::valueAt(size_t index)
+{
+	const auto buffer = valueBuffer();
+	return buffer ? buffer->data[index] : value();
```

Comment:
`data` is a method. You'd have to remember the `()`.

I would use:
```c++
return buffer ? (*buffer)[index] : value();
```

Reply:
ahh i (and saker) forgot the (), and i copy pasted him blindly

Reply:
Ah, apologies.

Reply:
That fail is now fixed, I get 
```sh
[ 38%] Building CXX object src/CMakeFiles/lmmsobjs.dir/core/AutomationClip.cpp.o
/home/roshan/storage/git/lmms/src/core/AutomatableModel.cpp: In member function ‘float lmms::AutomatableModel::valueAt(size_t)’:
/home/roshan/storage/git/lmms/src/core/AutomatableModel.cpp:332:49: error: no matching function for call to ‘lmms::AutomatableModel::value()’
  332 |         return buffer ? (*buffer)[index] : value();
      |                                            ~~~~~^~

``` 
now

Reply:
Try moving the `valueAt` function into `TypedAutomatableModel` instead.

In `AutomatableModel.h`:
```cpp
template <typename T> class LMMS_EXPORT TypedAutomatableModel : public AutomatableModel
{
public:
	using AutomatableModel::AutomatableModel;
	T value( int frameOffset = 0 ) const
	{
		return AutomatableModel::value<T>( frameOffset );
	}

       float AutomatableModel::valueAt(size_t index, int frameOffset = 0)
       {
              const auto buffer = valueBuffer();
              return buffer ? (*buffer)[index] : value(frameOffset);
       }
	...
};
```

(Added support for the `frameOffset` parameter just in case, though I don't remember seeing any functions that called `value` use this parameter).

Reply:
I beileve this will break something over here.

Reply:
What do you think will break?

Reply:
Most the models are `BoolModel`, `IntModel`, `FloatModel`, etc, which all inherit from `TypedAutomatableModel`, which inherits from `AutomatableModel`.

Reply:
Alrighty, that's what I missed. I didn't note the inheritance